### ISSUE TITLE
Dungeons found randomly not marked on region map

### ISSUE THREAD

**Issue by petchema** [Interkarma/daggerfall-unity#1435]
**Describe the bug**
Dungeons found by just traveling in the wilderness are not immediately marked on region map; Even by entering the dungeon.
I'm pretty sure it used to work.
Seen in stream https://www.youtube.com/watch?v=CSbxIeXyqc8&t=745
Dungeon marked after save/exit/reload: https://www.youtube.com/watch?v=CSbxIeXyqc8&t=868

**To Reproduce**
1. Approach a dungeon not already marked on the region map
2. Open region map with DUNGEONs enabled, verify that the dungeon pixel is not added
3. Enter then leave the dungeon
2. Open region map with DUNGEONs enabled, verify that the dungeon pixel is still not added

**Expected behavior**
Dungeon should be marked on map as soon as you approached it close enough

**Screenshots**
See video stream above around the marked timecode (12m 25s)

**Desktop (please complete the following information):**
 - Windows (I assume)
 - Version 0.10.2 (ssen on stream)

**Additional context**
He's using Tedious Travel that overloads the standard travel maps, could be the issue but I suspect it's not because it's not involved the proximity checks and the other dungeons display correctly.

**Comment by petchema** [comment]
Ok, it seems to be not entirely reliable; Next found dungeon marked correctly:
https://www.youtube.com/watch?v=CSbxIeXyqc8&t=959

**Comment by Interkarma** [comment]
I believe this has been resolved by changes db4e7d5, c93edc6, 5428b98, 9ac3b5a (all part of the same fix set).

What was happening is that events weren't being properly unregistered by a quest action, which over time would register too many delegates and crash event, which in turn blocked discovery. It was also filling up player logs with exception spam, which is how I located issue in first place.

I made several changes make discovery more robust and fix the delegates not unregistering properly.

I'm fairly confident this has been resolved now, so I'll close. It can be reopened later if problem seems to persist.
